File: 0066. Plus One/0066.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
 int* plusOne(int* digits, int digitsSize, int* returnSize) {
    // digits: integer array
    // digitsSize: size of that integer array
    // returnSize: size of the array that you are gonna return
    for (int i=digitsSize-1; i>=0; i--){
        if(digits[i]!=9){
            digits[i]++;
            *returnSize= digitsSize; // i've forgotten...
            return digits;
        }
        digits[i]=0;
    }
    // if all of them was 9, now all of them became 0
    // now create new array with the size of digitsSize+1
    int* newArr = (int*)malloc((digitsSize+1)*(sizeof(int)));
    newArr[0]=1;
    for(int i=0; i<digitsSize; i++) newArr[i+1]=digits[i];

    *returnSize = digitsSize+1;
    return newArr;
}

File: 0066. Plus One/0066.java
class Solution {
    public int[] plusOne(int[] digits) {
        // if last digit is not 9 (just increment by one)
        if(digits[digits.length-1]!=9){
            digits[digits.length-1]+=1;
            return digits;
        }

        // create new array for the last option and check if there is another number than 9
        int flag=0;
        int newArr[] = new int[digits.length+1];
        for (int i=0; i<digits.length; i++){
            newArr[i]=digits[i];
            if(digits[i]!=9) flag=1;
        }

        // if all digits are not 9 and last digit is 9 (check if there are multiple 9 at the end)
        if(flag==1){
            int j=1;
            while(digits[digits.length-j]==9){
                digits[digits.length-j]=0;
                j++;
            }
            digits[digits.length-j]+=1;
            return digits;
        }

        // if all digits are 9 (use the newArr that you created with the size of length+1)
        newArr[digits.length]=0;
        int j=1;
        while(digits.length-j!=0){
            newArr[digits.length-j]=0;
            j++;
        }
        newArr[digits.length-j]=1;
        return newArr;

    }
}